collector/perf_linux.go

```diff
@@ -49,23 +66,55 @@ func NewPerfCollector() (Collector, error) {
 		perfSwProfilers:    map[int]perf.SoftwareProfiler{},
 		perfCacheProfilers: map[int]perf.CacheProfiler{},
 	}
-	ncpus := runtime.NumCPU()
-	for i := 0; i < ncpus; i++ {
+
+	start := 0
+	ncpus := 0
+	var err error
+
+	if !isValidCPUString(cpus) {
```

Comment:
If the flag is set but it's invalid, it should fail instead of falling back to automatically figure this out.